doc/tutorials/virtualenvs.rst

```diff
@@ -0,0 +1,309 @@
+==========================================
+Creating and managing virtual environments
+==========================================
+
+When it comes to the management of third-party packages, Python
+has some complications:
+
+* By default, every third-party package will be installed to same directory.
+* Python is unable to differentiate between different versions of the same
+  package installed to that directory
+
+This means:
+
+* If you're working on a project and you make a backwards incompatible
+  change, then other projects that depend on it may be broken until you go
+  through and update them all with the necessary changes.
+* If you have two projects that depend on different version of the same
+  package then it becomes impossible for both to function simultaneously.
+
+In order to deal with these problems (and others) its recommended that you
+work within a Python `virtual environment
+<https://docs.python.org/3/tutorial/venv.html>`_, which is an
+isolated environment with its own set of installed system and third-party
```

Comment:
The venv link here documents Python's own ```python -m venv env-name``` option, so different than the `virtualenv`/`virtualenvwrapper` recommended below.  A matter of personal taste, obviously, but personally I think the `python -m venv` is easier, and requires no installation, but of course is Python 3 only (>=3.3 I believe). Perhaps it could be mentioned here a little more as another option.  Also, combined with the 'py launcher' it is quite easy to organize environments as well as use different Python versions easily.

Reply:
I've changed the link to the virtualenv docs and added a note at the bottom of the pip section about venv.